keps/sig-storage/20190129-csi-volume-resizing.md

```diff
@@ -130,6 +130,30 @@ If `NodeExpandVolume` is successful:
 If `NodeExpandVolume` failed:
 - It will add a event to both PVC and Pod about failed resizing and resize operation will be retried.
 
+#### Supporting per-PVC secret refs
+
+To support per-PVC secrets for volume resizing, similar to CSI attach and detach - this proposal expands `CSIPersistentVolumeSource` object to contain `ControllerExpansionSecretRef`:
+
+```
+type CSIPersistentVolumeSource struct {
+    ....
+    // ControllerPublishSecretRef is a reference to the secret object containing
+    // sensitive information to pass to the CSI driver to complete the CSI controller publish
+    ControllerPublishSecretRef *SecretReference
+
+    // ControllerExpansionSecretRef is a reference to secret object containing sensitive
+    // information to pass to the CSI driver to complete CSI controller expansion
+    ControllerExpansionSecretRef *SecretReference
```

Comment:
does this match the CSI operation name (e.g. "ControllerExpansionVolume") or should it be "ControllerExpandSecretRef"?

Reply:
we differentiate between controller and node-accessible secrets to avoid giving the kubelet cluster-wide storage credentials. do we need a `NodeExpandSecretRef` as well, which we would make accessible to the kubelet?

Reply:
We do not need that. Node expansion RPC call of CSI does not accept secrets.

Reply:
The CSI RPC call is  `ControllerExpandVolume`. It thought it will be bit mouthful to name the field `ControllerExpandVolumeSecrets` - but in retrospect it isn't too bad either. I will change.

Reply:
we drop the `...Volume` suffix for all other operation -> secret transformations (e.g. "NodeStageVolume" -> "NodeStageSecretRef")